pymeasure/instruments/hp/hp8657b.py

```diff
@@ -0,0 +1,190 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class HP8657B(Instrument):
+    """ Represents the Hewlett Packard 8657B signal generator
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        kwargs.setdefault('read_termination', '\r\n')
+        kwargs.setdefault('send_end', True)
+        super().__init__(
+            adapter,
+            "Hewlett-Packard HP8657B",
+            includeSCPI=False,
+            **kwargs,
+        )
```

Comment:
If the device does not support a different termination/send_end, I'd put these kwargs direct in the super.init:
```suggestion
    def __init__(self, adapter, **kwargs):
        super().__init__(
            adapter,
            "Hewlett-Packard HP8657B",
            includeSCPI=False,
            read_termination='\r\n',
            send_end=True,
            **kwargs,
        )
```